src/types/parse_mode.rs

```diff
@@ -74,6 +76,45 @@ pub enum ParseMode {
     Markdown,
 }
 
+impl TryFrom<&str> for ParseMode {
+    type Error = ();
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let normalized = value.to_lowercase();
+        match normalized.as_ref() {
+            "html" => Ok(ParseMode::HTML),
+            "markdown" => Ok(ParseMode::Markdown),
```

Comment:
Probably there is a need for `#[allow(deprecated)]` as `ParseMode::Markdown` is deprecated